projects/storefrontlib/src/cms-components/order-confirmation/components/order-confirmation-overview/order-confirmation-overview.component.ts

```diff
@@ -37,17 +37,164 @@ export class OrderConfirmationOverviewComponent implements OnInit, OnDestroy {
+      filter(() => Boolean(isoDate)),
+      map((textTitle) => {
+        const date = this.getDate(new Date(isoDate));
+
+        return {
+          title: textTitle,
+          text: [date],
+        };
+      })
+    );
+  }
+
+  getReplenishmentFrequencyCardContent(frequency: string): Observable<Card> {
+    return this.translation
+      .translate('checkoutOrderConfirmation.frequency')
+      .pipe(
+        filter(() => Boolean(frequency)),
+        map((textTitle) => ({
+          title: textTitle,
+          text: [frequency],
+        }))
+      );
+  }
+
+  getReplenishmentNextDateCardContent(isoDate: string): Observable<Card> {
+    return this.translation
+      .translate('checkoutOrderConfirmation.nextOrderDate')
+      .pipe(
+        filter(() => Boolean(isoDate)),
+        map((textTitle) => {
+          const date = this.getDate(new Date(isoDate));
+
+          return {
+            title: textTitle,
+            text: [date],
+          };
+        })
+      );
+  }
+
+  getOrderCodeCardContent(orderCode: string): Observable<Card> {
+    return this.translation
+      .translate('checkoutOrderConfirmation.orderNumber')
+      .pipe(
+        filter(() => Boolean(orderCode)),
+        map((textTitle) => ({
+          title: textTitle,
+          text: [orderCode],
+        }))
+      );
+  }
+
+  getOrderStatusCardContent(status: string): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutOrderConfirmation.status'),
+      this.translation.translate('orderDetails.statusDisplay', {
+        context: status,
+      }),
+    ]).pipe(
+      map(([textTitle, textStatus]) => ({
+        title: textTitle,
+        text: [textStatus],
+      }))
+    );
+  }
+
+  getPurchaseOrderNumber(poNumber: string): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutReview.poNumber'),
+      this.translation.translate('checkoutPO.noPoNumber'),
+    ]).pipe(
+      map(([textTitle, noneTextTitle]) => ({
+        title: textTitle,
+        text: [poNumber ? poNumber : noneTextTitle],
+      }))
+    );
+  }
+
+  getMethodOfPaymentCardContent(
+    hasPaymentInfo: PaymentDetails
+  ): Observable<Card> {
+    return combineLatest([
+      this.translation.translate('checkoutProgress.methodOfPayment'),
+      this.translation.translate('paymentTypes.paymentType_ACCOUNT'),
+      this.translation.translate('paymentTypes.paymentType_CARD'),
+    ]).pipe(
+      map(([textTitle, textAccount, textCard]) => ({
+        title: textTitle,
+        text: [Boolean(hasPaymentInfo) ? textCard : textAccount],
+      }))
+    );
+  }
+
+  getCostCenterCardContent(costCenter: CostCenter): Observable<Card> {
+    return this.translation.translate('checkoutPO.costCenter').pipe(
+      filter(() => Boolean(costCenter)),
+      map((textTitle) => ({
+        title: textTitle,
+        textBold: costCenter?.name,
+        text: ['(' + costCenter?.unit.name + ')'],
```

Comment:
Will there always be a unit? `costCenter?.unit?.name`

Reply:
will add just in case, but a cost center will always have a unit